File: backend/grafioschtrader-common/src/main/java/grafioschtrader/error/LimitEntityTransactionError.java
package grafioschtrader.error;

public class LimitEntityTransactionError {
  public final String entity;
  public final int limit;
  public final int transactionsCount;

  public LimitEntityTransactionError(String entity, int limit, int transactionsCount) {
    this.entity = entity;
    this.limit = limit;
    this.transactionsCount = transactionsCount;
  }
}

File: frontend/src/app/lib/login/service/limit.entity.transaction.error.ts
import {GetTransformedError} from './get.transformed.error';
import {TransformedError} from './transformed.error';

export class LimitEntityTransactionError implements GetTransformedError {
  public entity: string;
  public limit: number;
  public transactionsCount: number;

  getTransformedError(): TransformedError {
    const transformedError = new TransformedError();
    transformedError.errorClass = this;
    transformedError.msgKey = 'transactionDayLimit';
    transformedError.interpolateParams['limit'] = this.limit;
    transformedError.interpolateParams['i18nEntity'] = this.entity;

    return transformedError;
  }

}
